numpy/distutils/fcompiler/gnu.py

```diff
@@ -57,7 +57,7 @@ def gnu_version_match(self, version_string):
                     return ('gfortran', m.group(1))
         else:
             # Output probably from --version, try harder:
-            m = re.search(r'GNU Fortran\s+95.*?([0-9-.]+)', version_string)
+            m = re.search(r'95.*?([0-9-.]+)', version_string)
```

Comment:
What is actual compiler you use and the string you need to match here? Adding that string to the tested ones as a regression test would be good.